### PR TITLE
Baseline Test for Java Compute Discovery

### PR DESCRIPTION
**Pull request by andreamlin**
Add baseline test for DiscoGapic generated code for Compute API.

The baseline file is the concatenated output of the runDiscogapicGen gradle task.

### CODE REVIEW

**Comment by andreamlin** [comment]
Tests pass. PTAL

**Comment by garrettjonesgoogle** [comment]
Thanks for turning this around so quick!

The baseline file is unfortunately too large to manage. What we have done for testing of the proto3 codepath is that we have defined a fake service which exercises the functionality of codegen, but has no more than that. I would recommend coming up with a fake discovery doc service like that; maybe you could take compute and strip it down, or use another discovery doc, or type one up manually (any strategy works). 

Here is the fake proto3 service: https://github.com/googleapis/toolkit/blob/master/src/test/java/com/google/api/codegen/testdata/library.proto

You don't have to make yours as thorough quite yet since you're just bootstrapping.

**Comment by andreamlin** [comment]
Replaced large compute baseline with smaller cloudbilling API baseline.

File: src/main/java/com/google/api/codegen/discogapic/transformer/java/JavaDiscoGapicSchemaToViewTransformer.java [hunk 1]
```diff
@@ -151,6 +153,15 @@ private StaticLangApiMessageView generateSchemaClass(
           nameFormatter.publicMethodName(Name.from("set").join(propertyName)));
       properties.add(simpleProperty.build());
     }
+    // TODO(andrealin): implement sorting in a conventional way.
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/discogapic/transformer/java/JavaDiscoGapicSchemaToViewTransformer.java]: How is this unconventional?

File: src/test/java/com/google/api/codegen/DiscoGapicTestBase.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    config =
+        CodegenTestUtil.readConfig(
+            new SimpleDiagCollector(), getTestDataLocator(), gapicConfigFileNames);
+    if (config == null) {
+      return;
+    }
+  }
+
+  @Override
+  protected Object run() {
```
Comment (garrettjonesgoogle) [inline src/test/java/com/google/api/codegen/DiscoGapicTestBase.java]: If you return `Map<String, Doc>` like `GapicTestBase` ( https://github.com/andreamlin/toolkit/blob/112c8e4faeaa009b924fa4ca4be80a5843f9b5c8/src/test/java/com/google/api/codegen/GapicTestBase.java#L170 ), the baseline will have a separator line for each file that is generated (see https://github.com/googleapis/toolkit/blob/master/src/test/java/com/google/api/codegen/testdata/java_main_library.baseline for example), which makes the output easier to read.

File: src/main/java/com/google/api/codegen/discogapic/transformer/java/JavaDiscoGapicSchemaToViewTransformer.java (hunk 1 above)
Comment (andreamlin) [inline src/main/java/com/google/api/codegen/discogapic/transformer/java/JavaDiscoGapicSchemaToViewTransformer.java]: :D 
Comment removed.

File: src/test/java/com/google/api/codegen/DiscoGapicTestBase.java (hunk 2 above)
Comment (andreamlin) [inline src/test/java/com/google/api/codegen/DiscoGapicTestBase.java]: Done.

**Comment by andreamlin** [comment]
Fishing for comments! PTAL

**Comment by garrettjonesgoogle** [comment]
So unfortunately, the diff is showing all of the merged commits, which makes it hard to find your commits to review. I think I'll have to help you clean up the commit history so it can be reviewed properly.

**Comment by vchudnov-g** [comment]
LMK when the commit history has been cleaned and I'll review these changes.

**Comment by garrettjonesgoogle** [comment]
LGTM

**Comment by andreamlin** [comment]
Commit history cleaned up.

File: src/test/java/com/google/api/codegen/DiscoGapicTestBase.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+              null,
+              new LinkedList<String>());
+    } catch (IOException e) {
+      throw new IllegalArgumentException("Problem creating DiscoGapic generator.");
+    }
+
+    config =
+        CodegenTestUtil.readConfig(
+            new SimpleDiagCollector(), getTestDataLocator(), gapicConfigFileNames);
+    if (config == null) {
```
Comment (vchudnov-g) [inline src/test/java/com/google/api/codegen/DiscoGapicTestBase.java]: why do we need this `if` statement? You're returning anyway. Is there something  to fill in?

File: src/test/java/com/google/api/codegen/JavaDiscoGapicGeneratorTest.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
+/* Copyright 2016 Google Inc
```
Comment (vchudnov-g) [inline src/test/java/com/google/api/codegen/JavaDiscoGapicGeneratorTest.java]: fix date

File: src/test/java/com/google/api/codegen/DiscoGapicTestBase.java [hunk 5]
```diff
@@ -0,0 +1,147 @@
+/* Copyright 2016 Google Inc
```
Comment (vchudnov-g) [inline src/test/java/com/google/api/codegen/DiscoGapicTestBase.java]: fix date

File: src/test/java/com/google/api/codegen/DiscoGapicTestBase.java (hunk 3 above)
Comment (andreamlin) [inline src/test/java/com/google/api/codegen/DiscoGapicTestBase.java]: Done. 
Adding throwing exception if config is null.

File: src/test/java/com/google/api/codegen/DiscoGapicTestBase.java (hunk 5 above)
Comment (andreamlin) [inline src/test/java/com/google/api/codegen/DiscoGapicTestBase.java]: Done.

File: src/test/java/com/google/api/codegen/JavaDiscoGapicGeneratorTest.java (hunk 4 above)
Comment (andreamlin) [inline src/test/java/com/google/api/codegen/JavaDiscoGapicGeneratorTest.java]: Done.

**Pull request merged**